boot/boot.js

```diff
@@ -510,26 +537,46 @@ $tw.utils.evalGlobal = function(code,context,filename) {
 		contextValues.push(value);
 	});
 	// Add the code prologue and epilogue
-	code = "(function(" + contextNames.join(",") + ") {(function(){\n" + code + "\n;})();\nreturn exports;\n})\n";
+	code = [
+		(!$tw.browser ? polyfill : ""),
+		"(function(" + contextNames.join(",") + ") {",
+		"  (function(){\n" + code + "\n;})();",
+		(!$tw.browser && sandbox && !allowGlobals) ? globalCheck : "",
+		"  return exports;\n",
+		"})"
+	].join("\n");
+	// code = "(function(" + contextNames.join(",") + ") {(function(){  \n" + code + "\n;})();\nreturn exports;\n})\n";
 	// Compile the code into a function
 	var fn;
 	if($tw.browser) {
 		fn = window["eval"](code + "\n\n//# sourceURL=" + filename);
 	} else {
-		fn = vm.runInThisContext(code,filename);
+		if(sandbox){
+			fn = vm.runInContext(code,sandbox,filename)
+		} else {
+			fn = vm.runInThisContext(code,filename);
+		}
 	}
 	// Call the function and return the exports
 	return fn.apply(null,contextValues);
 };
-
+$tw.utils.sandbox = !$tw.browser ? vm.createContext({}) : undefined; 
 /*
 Run code in a sandbox with only the specified context variables in scope
 */
-$tw.utils.evalSandboxed = $tw.browser ? $tw.utils.evalGlobal : function(code,context,filename) {
-	var sandbox = $tw.utils.extend(Object.create(null),context);
-	vm.runInNewContext(code,sandbox,filename);
-	return sandbox.exports;
-};
+$tw.utils.evalSandboxed = $tw.browser ? $tw.utils.evalGlobal : function(code,context,filename,allowGlobals) {
```

Comment:
The handler function here has the arguments `code,context,filename,allowGlobals` but the definition of `$tw.utils.evalGlobal` above has `code,context,filename,sandbox,allowGlobals`.

This perhaps suggests that some more tests would be useful?

Reply:
evalSandboxed calls evalGlobal with the specified sandbox, and evalGlobal uses the sandbox if specified. I don't really know how or where to write tests. Check the code of evalSandboxed to see how the sandbox argument is used.

Reply:
AllowGlobals isn't checked unless sandbox is specified, and object assign should throw an error if sandbox is boolean, so I think that covers the possible variations introduced. Back then I thought you meant unit tests, but now I'm not sure if you meant making sure variables are correctly typed.